cmake/fake_tribits.cmake

```diff
@@ -288,10 +288,6 @@ ELSE()
     LIST(REMOVE_DUPLICATES PARSE_SOURCES)
   ENDIF()
   ADD_LIBRARY(${NAME} ${PARSE_SOURCES})
-  target_compile_options(
-    ${NAME}
-    PUBLIC $<$<COMPILE_LANGUAGE:CXX>:${KOKKOS_CXX_FLAGS}>
-  )
```

Comment:
Would need to discuss with @jjwilke and @masterleinad how this interacts with #2396

Reply:
I still need to delete the subsequent three lines for avoiding the duplicated flags.

Reply:
FYI, in the current develop branch, this variable KOKKOS_CXX_FLAGS only shows up in this location.  It is not set by KOKKOS_OPTION or explicitly, so the only way it would be set is through -DKOKKOS_CXX_FLAGS which is not desirable in our current mode of operation

Reply:
This is fine. Both `target_compile_options` and `target_link_options` should get deleted. Then we should add `target_compile_features` with `KOKKOS_CXX_STANDARD`.